dojo/tools/codechecker/parser.py

```diff
@@ -0,0 +1,128 @@
+import json
+import hashlib
+from dojo.models import Finding
+
+
+class CodeCheckerParser(object):
+
+    def get_scan_types(self):
+        return ["Codechecker Report native"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type  # no custom label for now
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import Codechecker Report in native JSON format."
+
+    def get_requires_file(self, scan_type):
+        return True
+
+    def get_findings(self, json_output, test):
+
+        if json_output is None:
+            return
+
+        tree = self.parse_json(json_output)
+        if tree:
+            return self.get_items(tree)
+
+    def parse_json(self, json_output):
+        data = json_output.read()
+        return json.loads(data)
+
+    def get_items(self, tree):
+        items = {}
+        # all findings are in "reports" list
+        for node in tree['reports']:
+            item = get_item(node)
+            if item:
+                items[item.unique_id_from_tool] = item
+
+        return list(items.values())
+
+
+def get_item(vuln):
+
+    description = 'Analyzer name: {}\n'.format(vuln['analyzer_name'])
+    description += 'Category name: {}\n'.format(vuln['category'])
+    description += 'Checker name: {}\n'.format(vuln['checker_name'])
+
+    if 'type' in vuln:
+        vuln_type = vuln.get('type', 'None')
+        if vuln_type != 'None':
+            description += 'Type: {}\n'.format(vuln_type)
+
+    if 'message' in vuln:
+        description += '{}\n'.format(vuln['message'])
+
+    location = vuln['file']
+    file_path = location['path'] if 'path' in location else None
+
+    if file_path:
+        description += 'File path: {}\n'.format(file_path)
+
+    line = vuln['line'] if 'line' in vuln else None
+    column = vuln['column'] if 'column' in vuln else None
+
+    if line is not None and column is not None:
+        description += 'Location in file: line {}, column {}\n'.format(line, column)
+
+    sast_source_line = line
+
+    sast_object = None
+
+    severity = get_mapped_severity(vuln.get('severity', 'UNSPECIFIED'))
+
+    mitigation = ''
+
+    references = ''
```

Comment:
Same comment as for `mitigation`. Make it simple. If we don't have references, just remove this part.